yt/utilities/answer_testing/utils.py

```diff
@@ -282,8 +279,92 @@ def _save_arrays(save_dir_name, fbasename, arrays, answer_store):
     answer_store : bool
         If true, save the just-generated test results, otherwise,
         compare them to the previously saved results.
+
+    func_name : str
+        The name of the function that created the answers being saved.
+    """
+    answer_file = os.path.join(save_dir_name, fbasename + '.h5')
+    if answer_store:
+        _save_raw_arrays(arrays, answer_file, func_name)
+    else:
+        _compare_raw_arrays(arrays, answer_file, func_name)
+
+
+def _save_raw_arrays(arrays, answer_file, func_name):
+    r"""
+    Saves the raw arrays produced from answer tests to a file.
+
+    The structure of `answer_file` is: each test function (e.g.,
+    test_toro1d[0-None-None-0]) forms a group. Within each group is a
+    hdf5 dataset named after the test (e.g., field_values). The value
+    stored in each dataset is the raw array corresponding to that
+    test and function.
+
+    Parameters
+    ----------
+    arrays : dict
+        Keys are the test name (e.g. field_values) and the values are
+        the actual answer arrays produced by the test.
+
+    answer_file : str
+        The name of the file to save the answers to, in hdf5 format.
+
+    func_name : str
+        The name of the function (possibly augmented by pytest with
+        test parameters) that called the test functions
+        (e.g, test_toro1d).
+    """
+    with h5py.File(answer_file, 'a') as f:
+        grp = f.create_group(func_name)
+        for test_name, test_array in arrays.items():
+            # Some answer tests (e.g., grid_values, projection_values)
+            # return a dictionary, which cannot be handled by h5py
+            if isinstance(test_array, dict):
+                test_array = _dict_to_array(test_array)
+            grp.create_dataset(test_name, data=test_array)
```

Comment:
I don't understand what this block is doing and how it interacts with the answer file.
Could you add some more comments to clarify this ?